src/main/resources/net/minecraftforge/gradle/GradleStartCommon.java

```diff
@@ -228,12 +276,33 @@ private void searchCoremods() throws Exception {
                     manifest = new Manifest(stream);
                     stream.close();
                 }
-            } else if (coreMod.getName().endsWith("jar")) // its a jar
-            {
+
+                // Try to detect directory grimpatch
+                for (File file : coreMod.listFiles()) {
+                    if (file.isFile() && this.isMixinConfiguration(file.getName())) {
```

Comment:
I can see this breaking on windows if it's using the coremod as file because of "\", but I never saw anyone using a unpacked coremod anyways.

Reply:
it's usually good to have config files inside a sub folder with the mod name to prevent conflicts, most of my complex patches does that because of how generic the names are. But I guess it will be really rare to get to that situation, if some issue arises later then this can be changed.